local/bin/py/build/process_agent_config.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+
+import json
+import re
+from os import getenv
+from collections import defaultdict
+
+def format_agent_config_string(string):
+  """
+  Takes a string from the agent config template and formats it for output in the agent_config shortcode.
+    - If string contains exactly one '#' and no alphabetic characters, remove the '#' and any new line characters.
+    - If the string contains exactly one '#' and has alphabetic characters, remove the '#' and leave the characters (this represents a config key/value)
+  """
+  # Match any lines containing strictly one '#' character that can have any number of leading or trailing whitespaces, and no words.  Matching lines in the config template file are for spacing only.
+  regex_string = r"[^a-zA-Z0-9#]*#[^a-zA-Z0-9#]*$"
+
+  if re.match(regex_string, string):
+    return string.replace('#', '').replace('\n', '').strip()
+  elif '#' in string and '##' not in string:
+    return string.replace('#', '') + '\n'
+  else:
+    return string + '\n'
+
+def create_agent_config_dict(dd_agent_config_string):
+  """
+  Returns a dictionary where the keys represent each config type (i.e. "Basic Configuration", "Log Collection Configuration", "DogstatsD Configuration", etc),
+  and the values are a string containing the config options and code comments.
+  """
+  config_type_header_delimiter = '#######'
+  agent_config_array = dd_agent_config_string.splitlines()
+  agent_config_dict = {}
+  current_config_type = ''
+
+  for index, line in enumerate(agent_config_array):
+    if config_type_header_delimiter in line:
+      config_type = agent_config_array[index + 1].replace('#', '').strip().lower()
+
+      if config_type != '':
```

Comment:
you can probably just do `if config_type` as empty string will evaluate to false